contributors/design-proposals/server-get.md

```diff
@@ -0,0 +1,170 @@
+  schemes for retrieving data from the server
+* Keep implementation of `get` and `describe` simple
+* Ease internationalization of `get` and `describe` output for all clients
+
+## Non-Goals
+
+* Deep customization of the returned output by the client
+
+
+## Specification of server-side `get`
+
+The server would return a `TableList` object (working-name) that contains metadata for columns and one or more
+rows composed of cells for each column.  Some additional data may be relevant for each row and returned by the
+server. Since every object should have a `TableList` representation, treat this as part of content negotiation
+as described in [the alternative representations of objects proposal](alternate-api-representations.md).
+
+Example request:
+
+```
+$ curl https://localhost:8443/api/v1/pods -H "Accept: application/json+vnd.kubernetes.as+meta.k8s.io+v1alpha1+TableList"
+{
+    "kind": "TableList",
+    "apiVersion": "meta.k8s.io/v1alpha1",
+    "headers": [
+        {"name": "Name", "type": "string", "description": "The name of the pod, must be unique ..."},
+        {"name": "Status", "type": "string", "description": "Describes the current state of the pod"},
+        ...
+    ],
+    "items": [
+        {"cells": ["pod1", "Failed - unable to start", ...]},
+        {"cells": ["pod2", "Init 0/2", ...]},
+        ...
+    ]
+}
+```
+
+This representation is also possible to return from a watch:
+
+```
+$ curl https://localhost:8443/api/v1/pods?watch=1 -H "Accept: application/json+vnd.kubernetes.as+meta.k8s.io+v1alpha1+TableList"
+{
+    "kind": "TableList",
+    "apiVersion": "meta.k8s.io/v1alpha1",
+    // headers are printed first, in case the watch holds
+    "headers": [
+        {"name": "Name", "type": "string", "description": "The name of the pod, must be unique ..."},
+        {"name": "Status", "type": "string", "description": "Describes the current state of the pod"},
+        ...
+    ]
+}
+{
+    "kind": "TableList",
+    "apiVersion": "meta.k8s.io/v1alpha1",
+    // headers are not returned here
+    "items": [
+        {"cells": ["pod1", "Failed - unable to start", ...]},
+        ...
+    ]
+}
+```
+
+It can also be returned in CSV form:
+
+```
+$ curl https://localhost:8443/api/v1/pods -H "Accept: text/csv+vnd.kubernetes.as+meta.k8s.io+v1alpha1+TableList"
+Name,Status,...
+pod1,"Failed - unable to start",...
+pod2,"Init 0/2",...
+...
+```
+
+To support "wide" format, columns may be marked with an optional priority field of increasing integers (default
+priority 0):
+
+```
+{
+    "kind": "TableList",
+    "apiVersion": "meta.k8s.io/v1alpha1",
+    "headers": [
+        ...
+        {"name": "Node Name", "type": "string", "description": "The node the pod is scheduled on, empty if the pod is not yet scheduled", "priority": 1},
+        ...
+    ],
+    ...
+}
+```
+
+To allow label columns, and to enable integrators to build effective UIs, each row may contain object metadata:
```

Comment:
This would be standard object metadata, OR it could be a runtime.Object and you have to ask for the object metadata.  Note that this object is in metav1 / metaalphav1, which is an API that evolves.  So until we break object meta, you can use this, and then in the future you can just ask for metav2.